lib/Controller/FolderController.php

```diff
@@ -84,10 +84,28 @@ public function getFolders(): DataResponse {
 	 * @RequireGroupFolderAdmin
 	 */
 	public function getFolder(int $id): DataResponse {
-		return new DataResponse($this->manager->getFolder($id, $this->getRootFolderStorageId()));
+		$response = $this->checkFolderExists($id);
+		if ($response) {
+			return $response;
+		}
+
+		$storageId = $this->getRootFolderStorageId();
+		return new DataResponse($this->manager->getFolder($id, $storageId));
+	}
+
+	private function checkFolderExists(int $id): ?DataResponse {
+		$storageId = $this->getRootFolderStorageId();
+		if ($storageId === null) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
```

Comment:
The class is not found (see the failing static analysis check) as the `use OCP\AppFramework\Http;` declaration is missing in the file (L28)